pkg/server/service.go

```diff
@@ -42,10 +51,30 @@ type CRIContainerdService interface {
 }
 
 // criContainerdService implements CRIContainerdService.
-type criContainerdService struct{}
+type criContainerdService struct {
+	// rootDir is the directory for managing cri-containerd files.
+	rootDir string
+	// containerService is containerd container service client.
+	containerService execution.ContainerServiceClient
+	// sandboxStore stores all sandbox metadata.
+	sandboxStore metadata.SandboxStore
+	// sandboxNameIndex stores all sandbox names and make sure each name
+	// is unique.
+	sandboxNameIndex *registrar.Registrar
+	// sandboxIDIndex is trie tree for partial id index.
+	sandboxIDIndex *truncindex.TruncIndex
+}
 
 // NewCRIContainerdService returns a new instance of CRIContainerdService
-func NewCRIContainerdService(conn *grpc.ClientConn) CRIContainerdService {
+func NewCRIContainerdService(conn *grpc.ClientConn, rootDir string) CRIContainerdService {
 	// TODO: Initialize different containerd clients.
-	return &criContainerdService{}
+	return &criContainerdService{
+		rootDir:          rootDir,
+		containerService: execution.NewContainerServiceClient(conn),
```

Comment:
Yeah, that's in plan but not in scope of the alpha release. Will add a TODO here.